### DISCUSSION TITLE
Change httpx proxy timeout?

### DISCUSSION THREAD

**Discussion by ghost** [searxng/searxng#1537] [Q&A] [answered]
I am using Searx over Tor locally and I keep getting a 

    httpx:ProxyError raised by client.py in handle _async_request

and soon after a ProxyTimeoutError: 

    Proxy connection timed out: 10. 

Maybe it is because my connection is slow at proxying at startup? How could I try to extend this 10 seconds timeout? Thanks.

**Comment by mrpaulblack** [comment]
Hi @JAAnon  ,
There is a timeout for all engines defined here: https://github.com/searxng/searxng/blob/master/searx/settings.yml#L130= . If its your instance you can increase the timeout to a greater number (if you have a slow network for example).

**Reply by ghost** [reply]
Thanks, actually I had already increased the request timeout for all the engines, but my problem happens at the very startup of the app, I get a proxy timeout that, if I'm not mistaken, has a reference to client.py and async http request.

**Comment by ghost** [comment, marked as answer]
@mrpaulblack I have finally found the culprit by inspecting the latest error log: line 176 in network/network.py 
`response = await client.get("https://check.torproject.org/api/ip", timeout=10)`
it looks like that for some reason my connection is a bit slower at boot and 10 seconds are not enough, because if I change the value to "None" I don't have any issue.
Of course, I have to edit the file again every time that I update searxng. @dalf , if a timeout is not really necessary in this case, could you please consider disabling it by default by the next release?
